mlflow/pyfunc/__init__.py

```diff
@@ -627,6 +628,12 @@ def load_model(
             f'Model does not have the "{FLAVOR_NAME}" flavor',
             RESOURCE_DOES_NOT_EXIST,
         )
+    if os.path.exists(os.path.join(local_path, FEATURE_SPEC_RELATIVE_FILE_PATH)):
+        raise MlflowException(
+            "Unable to load Feature Store model directly. "
+            "Use score_batch for offline predictions.",
+            BAD_REQUEST,
+        )
```

Comment:
What if folks are loading feature store models today, e.g. to explore the signature? This would be backwards incompatible. It's pretty confusing to have a model with the pyfunc flavor that you can't load with `mlflow.pyfunc.load_model()`.

I'd suggest not throwing and perhaps print a warning instead.

Reply:
This shouldn't be possible. The model can't be loaded anyway without the loader module, and would fail here: https://github.com/mlflow/mlflow/blob/1c30171e1ba3953556926cf7355d59ac05d1a23f/mlflow/pyfunc/__init__.py#L636
While it is possible to install the loader module outside Databricks to avoid this (as with the FS client), this is not recommended or supported.

Reply:
Does this warrant a warning even on databricks?

Reply:
+ Even on databricks, we don't allow installation of the FS client and the loader module at the same time

Reply:
So do we need this PR? I'd rather not wedge custom logic into `mlflow.pyfunc` for this case if we can avoid it

Reply:
Without this, the import failure `ImportError: databricks.feature_store.mlflow_model` would wrongly imply that the user should install it. We should have some up-front logic (whether error or not) to produce a more useful message

Reply:
Alternately, is there a better place for that sort of thing?

Reply:
Ok I'm now with you on warning instead of error. Since the current implementation doesn't take much other than the loader module into account, this would break the pyfunc scoring server in model serving.